core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/HDFSFileMoveAction.java

```diff
@@ -0,0 +1,153 @@
+
+      try {
+        if (!fileSystem.rename(source, dest)) {
+          if (!config.continueOnError) {
+            throw new IOException(String.format("Failed to rename file {} to {}", source.toString(), dest.toString()));
+          }
+          LOG.error("Failed to move file {} to {}", source.toString(), dest.toString());
+        }
+      } catch (IOException e) {
+        if (!config.continueOnError) {
+          throw e;
+        }
+        LOG.error("Failed to move file {} to {}", source.toString(), dest.toString(), e);
+      }
+      return;
+    }
+
+    //moving contents of directory
+    FileStatus[] listFiles;
+    if (config.fileRegex != null) {
+      PathFilter filter = new PathFilter() {
+        private final Pattern pattern = Pattern.compile(config.fileRegex);
+
+        @Override
+        public boolean accept(Path path) {
+          return pattern.matcher(path.getName()).matches();
+        }
+      };
+
+      listFiles = fileSystem.listStatus(source, filter);
+    } else {
+      listFiles = fileSystem.listStatus(source);
+    }
+
+    for (FileStatus file: listFiles) {
+      source = file.getPath();
+      //Moving directory, so destination filePath doesn't have filename
+      dest = new Path(config.destPath);
+
+      try {
+        if (!fileSystem.rename(source, dest)) {
+          if (!config.continueOnError) {
+            throw new IOException(String.format("Failed to rename file {} to {}", source.toString(), dest.toString()));
+          }
+          LOG.error("Failed to move file {} to {}", source.toString(), dest.toString());
+        }
+      } catch (IOException e) {
+        if (!config.continueOnError) {
+          throw e;
+        }
+        LOG.error("Failed to move file {} to {}", source.toString(), dest.toString(), e);
+      }
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+
+  }
+
+  /**
+   * Config class that contains all properties necessary to execute an HDFS move command.
+   */
+  public class HDFSActionConfig extends PluginConfig {
+    @Description("Full HDFS path of file/directory. In the case of a directory, if fileRegex is set, " +
+      "then all files in the directory matching the wildcard regex will be moved. Otherwise, all files in the " +
+      "directory will be moved. Ex: hdfs://hostname/tmp")
+    private String sourcePath;
+
+    @Description("The full HDFS destination path in the same cluster where user wants to move the file(s) in " +
+      "sourcePath. It is assumed that the path is valid. For moving a file, this means that the parent directories " +
+      "exist. For moving multiple files in a directory or an entire directory, the path to the desired directory " +
+      "location should be passed and that directory and all parent directories should already exist.")
+    private String destPath;
+
+    @Description("Wildcard regex to filter type of files in the directory to move")
```

Comment:
"Wildcard regular expression to filter the files in the directory that will be moved"